### ISSUE TITLE
BUG: If someone grabs quad in same mvd frame they die, pickup is not detected

### ISSUE THREAD

**Issue by meag** [ezQuake/ezquake-source#519]
**ezQuake version:**
All (up to 3.6-alpha7)

**Describe the bug**
Between mvd frames, the player grabs quad but also dies.
mvd stats code to detect pickups only runs for live players, so pickup isn't detected, client-side game stats etc incorrect.

**To Reproduce**
Attached .mvd [ztn.zip](https://github.com/ezQuake/ezquake-source/files/6014890/ztn.zip)
First quad pickup not detected

**Expected behavior**
Quad & ring pickups should be detected using KTX notifications if they are available (they are in example demo)
Attempt should also be made for older demos by checking newly-dead players for powerups/armors

**Comment by hemostx** [comment]
<q>Quad & ring pickups should be detected using KTX notifications if they are available </q>
We didn't quite get to notice it as blatanly in that demo, but could this also happen with *any* item besides powerups? There are stats for armours and weapons which could get misrepresented too.

**Comment by meag** [comment]
You're right hemo - I said powerups/armors in the next line but really everything should go through KTX notifications if they're active.
